tests/system/marathon_common_tests.py

```diff
@@ -1154,3 +1157,173 @@ def test_ipv6_healthcheck(docker_ipv6_network_fixture):
+
+    # Ensure that all instances are running in the same region, and on *all* the region zones
+    app = client.get_app(app_def['id'])
+    common.assert_app_in_all_domains(app, regions=(region,), zones=zones)
+
+    # Remove app
+    client.remove_app(app['id'], True)
+
+@shakedown.dcos_1_11
+@pytest.mark.skipif("shakedown.ee_version() is None")
+def test_faultdomains_region_and_zone():
+    """Tests if the applications with a `region` and a `zone` defined are always starting on the
+       same region and zone
+    """
+
+    # Find out the biggest region in the cluster
+    (region, zones) = get_biggest_cluster_region()
+
+    assert len(zones) > 1, "The tests require at least 2 zones in the biggest region (%s)" % (
+        region,
+    )
+
+    single_zone = zones[0]
+
+    # Create an app pinned on the given region and zone.
+    #
+    # (We are starting more instances than the number of zones in the region in order to increase
+    #  the chance of a faulty instance to appear on a different zone)
+    #
+    app_def = apps.faultdomain_app(region=region, zone=single_zone, instances=len(zones) * 2)
+    client = marathon.create_client()
+    client.add_app(app_def)
+
+    # Wait for the app to be deployed
+    shakedown.deployment_wait(timeout=timedelta(minutes=1).total_seconds(), app_id=app_def['id'])
+
+    # Ensure that all instances are running in the same region, and on *all* the region zones
+    app = client.get_app(app_def['id'])
+    common.assert_app_in_all_domains(app, regions=(region,), zones=(single_zone,))
+
+    # Remove app
+    client.remove_app(app['id'], True)
+
+@shakedown.dcos_1_11
+@pytest.mark.skipif("shakedown.ee_version() is None")
+def test_faultdomains_region_unique():
+    """Tests if the applications with a ["@region", "UNIQUE"] constraint correctly starts the apps
+       across all regions in the cluster
+    """
+
+    # Find out all the regions in the cluster
+    regions = set(get_all_cluster_regions().keys())
+
+    assert len(regions) > 1, "The tests require at least 2 regions in the cluster"
+
+    # Create an app that should be deployed on a unique region
+    # The number of instances should be the number of regions available in the system
+    app_def = apps.faultdomain_app(
+        instances=len(regions),
+        constraints=[["@region", "UNIQUE"]]
+    )
+    client = marathon.create_client()
+    client.add_app(app_def)
+
+    # Wait for the app to be deployed
+    shakedown.deployment_wait(timeout=timedelta(minutes=1).total_seconds(), app_id=app_def['id'])
+
+    # Check if the application was launched correctly
+    app = client.get_app(app_def['id'])
+    (used_regions, used_zones) = get_used_regions_and_zones(get_app_domains(app))
+
+    assert used_regions == set(regions), "App %s was not launched in all regions: %s instead of %s" % (
+            app['id'], ', '.join(used_regions), ', '.join(regions)
+        )
+
+    # Remove app
+    client.remove_app(app['id'], True)
```

Comment:
I thinks we have a fixture/cleanup for this.

Reply:
So, are we deleting all apps at teardown? Or is there something we can annotate the test with? Because I only saw tests deploying apps with `add_app` and nothing cleans them 🤔

Reply:
this should definitely NOT be in the test itself... if it isn't already handled by a test fixture... a test fixture should be created.